lib/src/test/kotlin/integration/provider/ProviderTest.kt

```diff
@@ -0,0 +1,352 @@
+package integration.provider
```

Comment:
these are tests not requiring gas, right? so not exactly provider tests per se. I'd rename filename to something more meaningful, like `TestsNotRequiringGas.kt`. Maybe we should rename other integration test files as well.

Reply:
As I mentioned [here](https://github.com/software-mansion/starknet-jvm/pull/281#discussion_r1296904503), the idea is that AccountTest, ProviderTest, etc. will eventually mostly mimic the test suites for devnet.
 
Whether the test case requires gas will be specified in the beginning of said test case.

Reply:
I don't necessarily think they should mimic that behaviour - Ideally we would have a simple test suite that tests the whole flow (or at least the most important parts of it, like declaring, deploying, invoking contracts + maybe getting some info from chain to verify transactions). I don't think it is a good idea to basically duplicate Account and Provider test suites for different chain type.